go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -124,9 +124,8 @@ func transformHorizon(ctx *plancontext.PlanningContext, op *operators.Horizon, i
 		}
 
 		return planLimit(node.Limit, plan)
-	default:
-		panic("only SELECT and UNION implement the SelectStatement interface")
 	}
+	panic("only SELECT and UNION implement the SelectStatement interface")
```

Comment:
nit: the existing function can return an error, so, should we replace `panic` with an `error`